prime-router/src/main/kotlin/history/db/ReportGraph.kt

```diff
@@ -66,47 +82,102 @@ class ReportGraph(
-        DSL.name(METADATA_CTE).`as`(
+    fun metadataCommonTableExpression(
+        itemGraphRecords: CommonTableExpression<ItemGraphRecord>
+    ): CommonTableExpression<CovidResultMetadataRecord> {
+        return DSL.name(METADATA_CTE).`as`(
             selectDistinct(COVID_RESULT_METADATA.asterisk())
                 .from(COVID_RESULT_METADATA)
                 .where(
                     COVID_RESULT_METADATA.REPORT_ID.`in`(
-                        selectDistinct(sourceReportIds.field(REPORT_FILE.REPORT_ID.name, SQLDataType.UUID)).from(
-                            sourceReportIds
+                        selectDistinct(
+                            DSL.field(
+                                ItemGraphTable.ITEM_GRAPH.PARENT_REPORT_ID.unqualifiedName,
+                                SQLDataType.UUID
+                            )
+                        ).from(
+                            itemGraphRecords
                         )
                     )
-                )
+                ).and(
+                    COVID_RESULT_METADATA.REPORT_INDEX.`in`(
+                        selectDistinct(
+                            DSL.field(ItemGraphTable.ITEM_GRAPH.PARENT_INDEX.unqualifiedName, SQLDataType.INTEGER)
+                        ).from(itemGraphRecords)
+                    )
+                ).coerce(COVID_RESULT_METADATA)
         )
+    }
 
     /**
-     * Accepts a walked graph of report ids and finds the corresponding report file filtering down to
-     * reports where the sending org is not null
+     * Accepts a list of report ids and then finds all the items associated with that report
+     * to then recursively walk the item lineage to produce an [ItemGraphRecord]
+     *
+     * Results will look like:
+     * 784f82f6-75f7-4ccc-aad1-1ab24ad8b595, 1, (784f82f6-75f7-4ccc-aad1-1ab24ad8b595,1)
+     * 81492979-40cd-45e7-a7ce-038269fd81aa, 1, (81492979-40cd-45e7-a7ce-038269fd81aa,1)
+     * 5a52b273-c79e-46fa-9c5a-65fa725d4daa, 1, (784f82f6-75f7-4ccc-aad1-1ab24ad8b595,1)->(5a52b273-c79e-46fa-9c5a-65fa725d4daa,1)
+     * 5a52b273-c79e-46fa-9c5a-65fa725d4daa, 2, (81492979-40cd-45e7-a7ce-038269fd81aa,1)->(5a52b273-c79e-46fa-9c5a-65fa725d4daa,2)
      *
-     * @param lineage a CTE that represents a graph of reports
+     * @param reportIds the set of reports to start from
      */
-    private fun sourceReportsCommonTableExpression(
-        lineage: CommonTableExpression<Record2<UUID, String>>
-    ): CommonTableExpression<Record> {
-        val sourceCteName = "source"
-        return DSL.name(sourceCteName).`as`(
-            DSL.select(REPORT_FILE.asterisk()).from(REPORT_FILE).join(lineage.name).on(
-                REPORT_FILE.REPORT_ID.eq(lineage.field(PARENT_REPORT_ID_FIELD, SQLDataType.UUID))
-            ).where(REPORT_FILE.SENDING_ORG.isNotNull)
-        )
-    }
+    fun itemAncestorGraphCommonTableExpression(reportIds: List<UUID>) =
```

Comment:
This is a new graph search using the item lineage table rather than just the report lineage table.  This is required to make sure that the returned metadata rows are only the ones that appear in the report sent to the receiver